src/braket/pennylane_plugin/braket_device.py

```diff
@@ -204,12 +214,28 @@ def apply(
 
         return circuit
 
+    def _check_supported_result_types(self):
+        try:
+            supported_result_types = self._device.properties.action[
+                "braket.ir.jaqcd.program"
+            ].supportedResultTypes
+        except AttributeError:
+            warnings.warn("Device does not support any gate-based result types")
+            self._braket_result_types = frozenset()
+            return
+
+        self._braket_result_types = frozenset(
+            result_type.name for result_type in supported_result_types
+        )
```

Comment:
Maybe I'm blind, but where is this called?

Reply:
Nice catch - it got lost in a merge. Fixing.